src/nodes/Components/MetaMask.js

```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
-            reject(error)
-          }else{
-            console.log("RESOLVE",result)
-            resolve(result.result)
-          }
-
+          params: [args.message, from],
+          from: from,
         })
-      });
-    }
-  })
+        .then((result) => {
+          console.dir(result)
+          console.log('PERSONAL SIGNED: ' + JSON.stringify(result))
+          console.log('recovering...')
+
+          const msgParams = { data: args.message, sig: result }
+          console.log(from.length)
+          const recovered = sigUtil.recoverPersonalSignature(msgParams)
+          console.dir({ recovered })
+
+          if (recovered === from ) {
+            console.log('SigUtil Successfully verified signer as ' + from)
+            resolve(window.alert('SigUtil Successfully verified signer as ' + from))
+          } else {
+            console.dir(recovered)
+            console.log('SigUtil Failed to verify signer when comparing ' + recovered.result + ' to ' + from)
+            reject(console.log('Failed, comparing %s to %s', recovered, from))
+          } 
+        }).catch((error) => {
+          return console.error(error)
+        })
+    })
+  }})
+
+
   this.setOutputData(3,{
     name:"send",
     args:[{name:"to",type:"string"},{name:"value",type:"number"},{name:"data",type:"string"},{name:"gasLimit",type:"number"},{name:"gasPrice",type:"number"}],
     function:async (args)=>{
-      return new Promise((resolve, reject) => {
+      await new Promise((resolve, reject) => {
         this.onAction()
-        let currentWeb3 = new Web3(window.web3)
+        let currentWeb3 = new Web3(window.ethereum)
 
         const transactionParameters = {
           //gasPrice: '0x09184e72a000', // customizable by user during MetaMask confirmation.
           //gas: '0x2710',  // customizable by user during MetaMask confirmation.
           //to: args.to, // Required except during contract publications.
-          from: window.ethereum.selectedAddress, // must match user's active address.
+          from: this.accounts[0], // must match user's active address.
           //value: ""+args.value, // Only required to send ether to the recipient from the initiating external account.
           //data: '0x7f7465737432000000000000000000000000000000000000000000000000000000600057', // Optional, but used for defining smart contract creation and interaction.
           //chainId: 3 // Used to prevent transaction reuse across blockchains. Auto-filled by MetaMask.
         }
 
-        if(typeof args.to != "undefined" && args.to!=null ){
+        if(typeof args.to !== "undefined" && args.to!=null ){
           transactionParameters.to = args.to
         }
 
-        if(typeof args.value != "undefined" && args.value!=null ){
-          transactionParameters.value = ""+currentWeb3.utils.toHex(args.value)
+        if(typeof args.value !== "undefined" && args.value!=null ){
+          let value = currentWeb3.utils.toWei(args.value.toString(), 'ether')
```

Comment:
Changed this input value to be in ether so easier to use than typing ether total in Wei.